packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -822,11 +822,20 @@ abstract class FlutterCommand extends Command<void> {
     if (devices.isEmpty && deviceManager.hasSpecifiedDeviceId) {
       globals.printStatus(userMessages.flutterNoMatchingDevice(deviceManager.specifiedDeviceId));
       return null;
-    } else if (devices.isEmpty && deviceManager.hasSpecifiedAllDevices) {
-      globals.printStatus(userMessages.flutterNoDevicesFound);
-      return null;
     } else if (devices.isEmpty) {
-      globals.printStatus(userMessages.flutterNoSupportedDevices);
+      if (deviceManager.hasSpecifiedAllDevices) {
+        globals.printStatus(userMessages.flutterNoDevicesFound);
+      } else {
+        globals.printStatus(userMessages.flutterNoSupportedDevices);
+      }
+      final List<Device> unSupportedDevices = await deviceManager.getDevices();
+      if (unSupportedDevices.isNotEmpty) {
+        globals.printStatus(userMessages.flutterFoundButUnSupportedDevices);
+        globals.printStatus('');
+        await Device.printDevices(unSupportedDevices);
+        globals.printStatus('');
+        globals.printStatus(userMessages.flutterMissPlatformProject(FlutterProject.current().directory.path));
```

Comment:
I like the idea of showing unsupported devices when no supported devices are found, but the "Your project is missing platform projects, run `flutter create`." message is far too strong, and will be alarming to the much more common scenario of people who are trying to debug their Android app just forgot to plug in their phone.

I think just mentioning that the discovered devices are unsupported will be a sufficient hint.

Reply:
Or we could change that message to something less alarming like:
```
If you would like your app to run on Android or web, consider running `flutter create .` to generate projects for these platforms.
```
where "<Android or web>" are examples generated based on the discovered but unsupported devices.

Reply:
Supported it. message alarm like: 

```
No supported devices connected.
The following devices were found, but are not supported by this project:

MIX 2      • 2e7847af   • android-arm64  • Android 9 (API 28)
Web Server • web-server • web-javascript • Flutter Tools
Chrome     • chrome     • web-javascript • Google Chrome 81.0.4044.138

Your project /Users/liufengkai/Documents/Code/FlutterSupport/a_test is missing some platform projects.
If you would like your app to run on android or web, consider running `flutter create .` to generate projects for these platforms.
```

<android or web> generated by unsupported devices.